File: OpenValidationFramework-CSharp/OpenValidationFramework-CSharp/Validator/Variables/ValidatorVariableArrayContainingStaticPropertyVariable.cs
using OpenValidationFramework_CSharp;
using OpenValidationFramework_CSharp.Model;
using OpenValidationFramework_CSharp.Model.Primitive;
using System;
using System.Collections.Generic;
using System.Text;

namespace io.openvalidation.integration.Variables
{
    public class ValidatorVariableArrayContainingStaticPropertyVariable
    {
        public HUMLFramework huml;

        public ValidatorVariableArrayContainingStaticPropertyVariable()
        {
            huml = new HUMLFramework();

            var nameVar = huml.CreateVariable("nameVar", (DefaultModel model) => "Peter"); 

            var nameArray = huml.CreateVariable("nameArray", (DefaultModel model) => huml.CREATE_ARRAY("Hans", model.Name, nameVar.GetValue(model)));

            huml.appendRule("",
                new String[] { "Name" },
                "The profession should not be one of nameArray",
                (DefaultModel model) => huml.ONE_OF(model.Profession, nameArray.GetValue(model)),
                false
            );
        }

        public String ValidatorID
        {
            get { return ""; }
        }

        public HUMLFramework.OpenValidationSummary Validate(DefaultModel model)
        {
            return huml.validate(model);
        }

        public HUMLFramework.OpenValidationSummary Validate(Object model)
        {
            return this.Validate((DefaultModel)model);
        }
    }
}

File: OpenValidationFramework-Java/src/main/java/org/bag/openvalidation/validator/ValidatorVariableArrayContainingStaticPropertyVariable.java
package org.bag.openvalidation.validator;

import org.bag.model.StandardModel;
import org.bag.openvalidation.HUMLFramework;

public class ValidatorVariableArrayContainingStaticPropertyVariable implements HUMLFramework.IOpenValidator
{
   public HUMLFramework huml;

   public ValidatorVariableArrayContainingStaticPropertyVariable() {}

   public static ValidatorVariableArrayContainingStaticPropertyVariable createValidatorWithIntegers(){
      ValidatorVariableArrayContainingStaticPropertyVariable validator = new ValidatorVariableArrayContainingStaticPropertyVariable();
      validator.huml = new HUMLFramework();


      HUMLFramework.Variable numVar = validator.huml.CreateVariable("numVar", (StandardModel model) -> 3);

      HUMLFramework.Variable numbers = validator.huml.CreateVariable("numbers", (StandardModel model) -> validator.huml.CREATE_ARRAY(1,model.getIntegerValue1(),numVar.GetValue(model)));

      validator.huml.appendRule("",
              new String[]{ "inputNumber" },
              "the inputNumber must be one of numbers",
              (StandardModel model) -> validator.huml.NONE_OF(model.getBigDecimal1(), numbers.GetValue(model)),
              false
      );

      return validator;
   }

   public static ValidatorVariableArrayContainingStaticPropertyVariable createValidatorWithIntegersAndDoubles(){
      ValidatorVariableArrayContainingStaticPropertyVariable validator = new ValidatorVariableArrayContainingStaticPropertyVariable();
      validator.huml = new HUMLFramework();


      HUMLFramework.Variable numVar = validator.huml.CreateVariable("numVar", (StandardModel model) -> 3.3);

      HUMLFramework.Variable numbers = validator.huml.CreateVariable("numbers", (StandardModel model) -> validator.huml.CREATE_ARRAY(1.1,model.getIntegerValue1(),numVar.GetValue(model)));

      validator.huml.appendRule("",
              new String[]{ "inputNumber" },
              "the inputNumber must be one of numbers",
              (StandardModel model) -> validator.huml.NONE_OF(model.getBigDecimal1(), numbers.GetValue(model)),
              false
      );

      return validator;
   }

   public String getValidatorID(){
      return "";
   }

   public HUMLFramework.OpenValidationSummary validate(StandardModel model) {
      return huml.validate(model);
   }

   @Override
   public HUMLFramework.OpenValidationSummary validate(Object model) {
      return this.validate((StandardModel)model);
   }


}
